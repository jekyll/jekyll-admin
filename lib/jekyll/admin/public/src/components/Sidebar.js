import React from 'react';
import { Link } from 'react-router';

import styles from '../styles/sidebar.scss';

const Sidebar = () => (
  <div className="sidebar-container">
    <Link className="logo" to={`/`} />
    <div className="links-container">
      <ul>
        <li>
          <Link activeClassName="active" to={`/configuration`}><i className="fa fa-cog"></i>Configuration</Link>
        </li>
        <li>
          <Link activeClassName="active" to={`/pages`}><i className="fa fa-file-text"></i>Pages</Link>
        </li>
        <li>
          <Link activeClassName="active" to={`/posts`}><i className="fa fa-thumb-tack"></i>Posts</Link>
        </li>
        <li>
          <Link activeClassName="active" to={`/collections`}><i className="fa fa-book"></i>Collections</Link>
        </li>
        <div className="splitter"></div>
        <li>
          <Link activeClassName="active" to={`/data`}><i className="fa fa-database"></i>Data</Link>
        </li>
        <li>
          <Link activeClassName="active" to={`/static-files`}><i className="fa fa-file"></i>Static Files</Link>
        </li>
        <li>
          <Link activeClassName="active" to={`/plugins`}><i className="fa fa-plug"></i>Plugins</Link>
        </li>
        <li>
          <Link activeClassName="active" to={`/git`}><i className="fa fa-git-square"></i>Git</Link>
        </li>
      </ul>
    </div>
  </div>
);

export default Sidebar;
