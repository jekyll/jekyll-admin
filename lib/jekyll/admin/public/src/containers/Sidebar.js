import React, { Component, PropTypes } from 'react';
import { connect } from 'react-redux';
import { bindActionCreators } from 'redux';
import { Link } from 'react-router';
import _ from 'underscore';

// Components
import Splitter from '../components/Splitter';

// Actions
import { fetchCollections } from '../actions/collections';

export class Sidebar extends Component {

  componentDidMount() {
    const { fetchCollections } = this.props;
    fetchCollections();
  }

  renderCollections() {
    const { collections } = this.props;
    if (collections.length) {
      return _.chain(collections)
      .filter(col => col.title != 'Posts') // TODO
      .map((col, i) => {
        return (
          <li key={i}>
            <Link activeClassName="active" to={`/collections${col.path}`}>
              <i className="fa fa-book"></i>{col.title}
            </Link>
          </li>
        );
      }).value();
    }
  }

  render() {
    return (
      <div className="sidebar">
        <Link className="logo" to={`/pages`} />
        <div className="routes">
          <ul>
            <li>
              <Link activeClassName="active" to={`/pages`}><i className="fa fa-file-text"></i>Pages</Link>
            </li>
            <li>
              <Link activeClassName="active" to={`/collections/posts`}><i className="fa fa-thumb-tack"></i>Posts</Link>
            </li>

            {this.renderCollections()}

            <Splitter />
            <li>
              <Link activeClassName="active" to={`/data`}><i className="fa fa-database"></i>Data</Link>
            </li>
            <li>
              <Link activeClassName="active" to={`/static-files`}><i className="fa fa-file"></i>Static Files</Link>
            </li>
            <li>
              <Link activeClassName="active" to={`/plugins`}><i className="fa fa-plug"></i>Plugins</Link>
            </li>
            <li>
              <Link activeClassName="active" to={`/git`}><i className="fa fa-git-square"></i>Git</Link>
            </li>
            <Splitter />
            <li>
              <Link activeClassName="active" to={`/configuration`}><i className="fa fa-cog"></i>Configuration</Link>
            </li>
          </ul>
        </div>
      </div>
    );
  }
}

Sidebar.propTypes = {
  collections: PropTypes.array.isRequired,
  fetchCollections: PropTypes.func.isRequired
};

function mapStateToProps(state) {
  const { collections } = state;
  return {
    collections: collections.collections
  };
}

function mapDispatchToProps(dispatch) {
  return bindActionCreators({
    fetchCollections
  }, dispatch);
}

export default connect(
  mapStateToProps,
  mapDispatchToProps,
  null,
  {pure:false})(Sidebar); // fix this when react-router 3.0.0
