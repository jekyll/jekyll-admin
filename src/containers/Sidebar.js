import React, { Component, PropTypes } from 'react';
import { connect } from 'react-redux';
import { bindActionCreators } from 'redux';
import { Link } from 'react-router';
import { ADMIN_PREFIX } from '../constants';
import Splitter from '../components/Splitter';
import { fetchCollections } from '../actions/collections';
import { capitalize } from '../utils/helpers';
import { sidebar as SidebarTranslations } from '../constants/lang';
import _ from 'underscore';

export class Sidebar extends Component {

  componentDidMount() {
    const { fetchCollections } = this.props;
    fetchCollections();
  }

  renderCollections(hiddens = []) {
    const { collections } = this.props;

    if (!collections.length) {
      return null;
    }

    return _.map(collections, (col, i) => {
      if (_.indexOf(hiddens, col.label) == -1) {
        return (
          <li key={i}>
            <Link activeClassName="active" to={`${ADMIN_PREFIX}/collections/${col.label}`}>
              <i className="fa fa-book" />{capitalize(col.label)}
            </Link>
          </li>
        );
      }
    });
  }

  render() {
    const { config } = this.props;

    const defaults = {
      pages: {
        icon: 'file-text',
        link: 'pages',
        translation: 'pages'
      },
      datafiles: {
        icon: 'database',
        link: 'datafiles',
        translation: 'datafiles',
        splitterBefore: true
      },
      staticfiles: {
        icon: 'file',
        link: 'staticfiles',
        translation: 'staticfiles'
      },
      configuration: {
        icon: 'cog',
        link: 'configuration',
        translation: 'configuration',
        splitterBefore: true
      },
    };

    const defaultLinks = _.keys(defaults);
    let hiddenLinks;
    try {
      hiddenLinks = config.jekyll_admin.hidden_links;
    } catch (e) {
      hiddenLinks = [];
    }

    const visibleLinks = _.difference(defaultLinks, hiddenLinks);

    const links = [];
    _.each(visibleLinks, (link, index, list) => {
      const current = defaults[link];
      if (current.splitterBefore) {
        links.push(<Splitter key={'splitter'+index} />);
      }
      links.push(
        <li key={index}>
          <Link activeClassName="active" to={`${ADMIN_PREFIX}/${current.link}`}>
            <i className={`fa fa-${current.icon}`} />
            {SidebarTranslations[current.translation]}
          </Link>
        </li>
      );
    });

    return (
      <div className="sidebar">
        <Link className="logo" to={`${ADMIN_PREFIX}/pages`} />
        <ul className="routes">
          {this.renderCollections(hiddenLinks)}
          {links}
        </ul>
      </div>
    );
  }
}

Sidebar.propTypes = {
  collections: PropTypes.array.isRequired,
  fetchCollections: PropTypes.func.isRequired,
  config: PropTypes.object.isRequired,
};

const mapStateToProps = (state) => ({
  collections: state.collections.collections
});

const mapDispatchToProps = (dispatch) => bindActionCreators({
  fetchCollections
}, dispatch);

export default connect(mapStateToProps,mapDispatchToProps)(Sidebar);
