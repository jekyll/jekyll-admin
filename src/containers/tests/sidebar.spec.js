import React from 'react';
import _ from 'underscore';
import { mount } from 'enzyme';
import { Link } from 'react-router';
import { Sidebar } from '../Sidebar';

import { collections, config } from './fixtures';

const defaultProps = {
  config,
  collections,
};

const nonCollectionLinks = ['pages', 'datafiles', 'staticfiles', 'configuration'];

function setup(props = defaultProps) {
  const actions = {
    fetchCollections: jest.fn()
  };

  const component = mount(<Sidebar {...props} {...actions} />);

  return {
    component: component,
    actions: actions,
    links: component.find('.routes').find(Link)
  };
}

describe('Containers::Sidebar', () => {
  it('should render correctly', () => {
    const { links, component } = setup();
    const actual = links.length;
    const expected = nonCollectionLinks.length + component.prop('collections').length;

    expect(actual).toEqual(expected);
  });

  it('should not render hidden links', () => {
    const config_with_hidden_links = _.extend(config, {
      jekyll_admin: {
        hidden_links: [
          'posts',
          'pages'
        ]
      }
    });

    const { component, links } = setup(Object.assign({}, defaultProps, {
      config: config_with_hidden_links
    }));

    const actual = links.length;
    const expected =
      (nonCollectionLinks.length - 1) + (component.prop('collections').length - 1);
    expect(actual).toEqual(expected);
  });

  it('should call fetchCollections action after mounted', () => {
    const { actions } = setup();
    expect(actions.fetchCollections).toHaveBeenCalled();
  });

  it('should render fine with zero collections', () => {
    const { component, links, actions } = setup(Object.assign({}, defaultProps, {
      collections: [],
      config: {
        jekyll_admin: {}
      }
    }));
    expect(links.length).toEqual(4);
  });
});
