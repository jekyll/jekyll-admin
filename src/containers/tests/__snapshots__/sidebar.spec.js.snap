// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Containers::Sidebar does not render hidden links 1`] = `
<div
  className="sidebar"
>
  <a
    className="logo"
    onClick={[Function]}
    style={Object {}}
  />
  <ul
    className="routes"
  >
    <div
      className="accordion collapsed"
      onClick={[Function]}
      style={
        Object {
          "maxHeight": 50,
        }
      }
    >
      <div
        className="accordion-label"
      >
        <i
          aria-hidden="true"
          className="fa fa-briefcase"
        />
        Collections
        <div
          className="indicator"
        >
          <div
            className="counter"
          >
            1
          </div>
        </div>
      </div>
      <ul>
        <li>
          <a
            onClick={[Function]}
            style={Object {}}
          >
            <i
              aria-hidden="true"
              className="fa fa-book"
            />
            Movies
          </a>
        </li>
      </ul>
    </div>
    <div
      className="splitter"
    />
    <div
      className="splitter"
    />
    <li>
      <a
        onClick={[Function]}
        style={Object {}}
      >
        <i
          aria-hidden="true"
          className="fa fa-database"
        />
        Data Files
      </a>
    </li>
    <li>
      <a
        onClick={[Function]}
        style={Object {}}
      >
        <i
          aria-hidden="true"
          className="fa fa-file"
        />
        Static Files
      </a>
    </li>
    <div
      className="splitter"
    />
    <li>
      <a
        onClick={[Function]}
        style={Object {}}
      >
        <i
          aria-hidden="true"
          className="fa fa-cog"
        />
        Configuration
      </a>
    </li>
  </ul>
</div>
`;

exports[`Containers::Sidebar renders correctly 1`] = `
<div
  className="sidebar"
>
  <a
    className="logo"
    onClick={[Function]}
    style={Object {}}
  />
  <ul
    className="routes"
  >
    <div
      className="accordion collapsed"
      onClick={[Function]}
      style={
        Object {
          "maxHeight": 50,
        }
      }
    >
      <div
        className="accordion-label"
      >
        <i
          aria-hidden="true"
          className="fa fa-briefcase"
        />
        Collections
        <div
          className="indicator"
        >
          <div
            className="counter"
          >
            1
          </div>
        </div>
      </div>
      <ul>
        <li>
          <a
            onClick={[Function]}
            style={Object {}}
          >
            <i
              aria-hidden="true"
              className="fa fa-book"
            />
            Movies
          </a>
        </li>
      </ul>
    </div>
    <li>
      <a
        onClick={[Function]}
        style={Object {}}
      >
        <i
          aria-hidden="true"
          className="fa fa-book"
        />
        Posts
      </a>
    </li>
    <div
      className="splitter"
    />
    <li>
      <a
        onClick={[Function]}
        style={Object {}}
      >
        <i
          aria-hidden="true"
          className="fa fa-file-text"
        />
        Pages
      </a>
    </li>
    <div
      className="splitter"
    />
    <li>
      <a
        onClick={[Function]}
        style={Object {}}
      >
        <i
          aria-hidden="true"
          className="fa fa-database"
        />
        Data Files
      </a>
    </li>
    <li>
      <a
        onClick={[Function]}
        style={Object {}}
      >
        <i
          aria-hidden="true"
          className="fa fa-file"
        />
        Static Files
      </a>
    </li>
    <div
      className="splitter"
    />
    <li>
      <a
        onClick={[Function]}
        style={Object {}}
      >
        <i
          aria-hidden="true"
          className="fa fa-cog"
        />
        Configuration
      </a>
    </li>
  </ul>
</div>
`;

exports[`Containers::Sidebar renders with custom collection labels 1`] = `
<div
  className="sidebar"
>
  <a
    className="logo"
    onClick={[Function]}
    style={Object {}}
  />
  <ul
    className="routes"
  >
    <div
      className="accordion collapsed"
      onClick={[Function]}
      style={
        Object {
          "maxHeight": 50,
        }
      }
    >
      <div
        className="accordion-label"
      >
        <i
          aria-hidden="true"
          className="fa fa-briefcase"
        />
        Collections
        <div
          className="indicator"
        >
          <div
            className="counter"
          >
            2
          </div>
        </div>
      </div>
      <ul>
        <li>
          <a
            onClick={[Function]}
            style={Object {}}
          >
            <i
              aria-hidden="true"
              className="fa fa-book"
            />
            Tutorials
          </a>
        </li>
        <li>
          <a
            onClick={[Function]}
            style={Object {}}
          >
            <i
              aria-hidden="true"
              className="fa fa-book"
            />
            Docs
          </a>
        </li>
      </ul>
    </div>
    <li>
      <a
        onClick={[Function]}
        style={Object {}}
      >
        <i
          aria-hidden="true"
          className="fa fa-book"
        />
        Posts
      </a>
    </li>
    <div
      className="splitter"
    />
    <li>
      <a
        onClick={[Function]}
        style={Object {}}
      >
        <i
          aria-hidden="true"
          className="fa fa-file-text"
        />
        Pages
      </a>
    </li>
    <div
      className="splitter"
    />
    <li>
      <a
        onClick={[Function]}
        style={Object {}}
      >
        <i
          aria-hidden="true"
          className="fa fa-database"
        />
        Data Files
      </a>
    </li>
    <li>
      <a
        onClick={[Function]}
        style={Object {}}
      >
        <i
          aria-hidden="true"
          className="fa fa-file"
        />
        Static Files
      </a>
    </li>
    <div
      className="splitter"
    />
    <li>
      <a
        onClick={[Function]}
        style={Object {}}
      >
        <i
          aria-hidden="true"
          className="fa fa-cog"
        />
        Configuration
      </a>
    </li>
  </ul>
</div>
`;

exports[`Containers::Sidebar renders with zero collections 1`] = `
<div
  className="sidebar"
>
  <a
    className="logo"
    onClick={[Function]}
    style={Object {}}
  />
  <ul
    className="routes"
  >
    <li>
      <a
        onClick={[Function]}
        style={Object {}}
      >
        <i
          aria-hidden="true"
          className="fa fa-book"
        />
        Posts
      </a>
    </li>
    <div
      className="splitter"
    />
    <li>
      <a
        onClick={[Function]}
        style={Object {}}
      >
        <i
          aria-hidden="true"
          className="fa fa-file-text"
        />
        Pages
      </a>
    </li>
    <div
      className="splitter"
    />
    <li>
      <a
        onClick={[Function]}
        style={Object {}}
      >
        <i
          aria-hidden="true"
          className="fa fa-database"
        />
        Data Files
      </a>
    </li>
    <li>
      <a
        onClick={[Function]}
        style={Object {}}
      >
        <i
          aria-hidden="true"
          className="fa fa-file"
        />
        Static Files
      </a>
    </li>
    <div
      className="splitter"
    />
    <li>
      <a
        onClick={[Function]}
        style={Object {}}
      >
        <i
          aria-hidden="true"
          className="fa fa-cog"
        />
        Configuration
      </a>
    </li>
  </ul>
</div>
`;
